src/transformers/utils/hub.py

```diff
@@ -838,22 +836,32 @@ def push_to_hub(
                 )
             token = use_auth_token
 
-        if "repo_path_or_name" in deprecated_kwargs:
+        repo_path_or_name = deprecated_kwargs.pop("repo_path_or_name", None)
+        if repo_path_or_name is not None:
             warnings.warn(
                 "The `repo_path_or_name` argument is deprecated and will be removed in v5 of Transformers. Use "
-                "`repo_id` instead."
+                "`repo_id` instead.",
+                FutureWarning,
             )
-            repo_id = deprecated_kwargs.pop("repo_path_or_name")
+            if repo_id is not None:
+                raise ValueError(
+                    "`repo_id` and `repo_path_or_name` are both specified. Please set only the argument `repo_id`."
+                )
+            if os.path.isdir(repo_path_or_name):
+                # repo_path: infer repo_id from the path
+                repo_id = repo_id.split(os.path.sep)[-1]
+                working_dir = repo_id
+            else:
+                # repo_name: use it as repo_id
+                repo_id = repo_path_or_name
+                working_dir = repo_id.split("/")[-1]
+        else:
+            working_dir = repo_id.split("/")[-1]
+
         # Deprecation warning will be sent after for repo_url and organization
         repo_url = deprecated_kwargs.pop("repo_url", None)
         organization = deprecated_kwargs.pop("organization", None)
 
-        if os.path.isdir(repo_id):
-            working_dir = repo_id
-            repo_id = repo_id.split(os.path.sep)[-1]
-        else:
-            working_dir = repo_id.split("/")[-1]
-
```

Comment:
I have some difficulty to follow the issue and the fix in this PR.

So the issue happens if `organization` is included in `repo_id` (so `organization` here is `None`), and there is a local directory with the same name (path)?

If this is the case, I don't see how the fix works (sorry):

There is still `repo_id = repo_id.split(os.path.sep)[-1]`, and `organization = deprecated_kwargs.pop("organization", None)` is still `None` when passing to `repo_id = self._create_repo`.

The `repo_id` in the following call will be just the last part of the local path

`url = create_repo(repo_id=repo_id, token=token, private=private, exist_ok=True)`

so what would be `url.repo_id` here?

I am sure you know much better than me - it's just that I don't follow well and can't give feedback (if any).

Might be great if you can explain a bit 🤗

Reply:
Yes, thank you @Wauplin .

(I still think this is not obvious, and some comments along the code changes would help the review. But this my own taste of doing things, take it just a kind suggestion - for future PRs of course - if you think it makes sense).

(My reasoning capability isn't very fast 😅 )

Reply:
Always good to ask :) Having to explain it made me realize how blurry it was. I added 2 comments in the code :)